bindings/rust/wasmedge-sys/src/store.rs

```diff
@@ -446,59 +470,69 @@ impl Store {
-        unsafe { wasmedge::WasmEdge_StoreListGlobalRegisteredLength(self.ctx, mod_name.into()) }
+        let mod_name: WasmEdgeString = mod_name.as_ref().into();
+        unsafe { wasmedge::WasmEdge_StoreListGlobalRegisteredLength(self.ctx, mod_name.as_raw()) }
     }
 
-    /// Returns an iterator of names of all exported [globals](crate::Global) in the registered module.
+    /// Returns the names of all exported [globals](crate::Global) in the registered module.
     ///
     /// # Argument
     ///
     /// - `mod_name` specifies the name of the registered module.
-    pub fn reg_global_names_iter(&self, mod_name: impl AsRef<str>) -> Vec<String> {
-        let mut names: Vec<String> = vec![];
+    pub fn reg_global_names(&self, mod_name: impl AsRef<str>) -> Option<Vec<String>> {
         let len_global_names = self.reg_global_len(mod_name.as_ref());
-        if len_global_names > 0 {
-            let mut global_names = Vec::with_capacity(len_global_names as usize);
-            unsafe {
-                wasmedge::WasmEdge_StoreListGlobalRegistered(
-                    self.ctx,
-                    mod_name.into(),
-                    global_names.as_mut_ptr(),
-                    len_global_names,
-                );
-                global_names.set_len(len_global_names as usize);
-            }
-
-            for global_name in global_names {
-                let slice = unsafe { std::ffi::CStr::from_ptr(global_name.Buf as *const _) };
-                names.push(slice.to_string_lossy().into_owned());
+        match len_global_names > 0 {
+            true => {
+                let mut global_names = Vec::with_capacity(len_global_names as usize);
+                let mod_name: WasmEdgeString = mod_name.as_ref().into();
+                unsafe {
+                    wasmedge::WasmEdge_StoreListGlobalRegistered(
+                        self.ctx,
+                        mod_name.as_raw(),
+                        global_names.as_mut_ptr(),
+                        len_global_names,
+                    );
+                    global_names.set_len(len_global_names as usize);
+                }
+
+                let names = global_names
+                    .into_iter()
+                    .map(|x| x.into())
+                    .collect::<Vec<String>>();
+                Some(names)
             }
+            false => None,
         }
-        names
     }
 
     /// Returns the length of the exported [memories](crate::Memory) in the anonymous module.
     pub fn mem_len(&self) -> u32 {
         unsafe { wasmedge::WasmEdge_StoreListMemoryLength(self.ctx) }
     }
 
-    /// Returns an iterator of names of all exported [memories](crate::Memory) in the anonymous module.
-    pub fn mem_names_iter(&self) -> Vec<String> {
-        let mut names: Vec<String> = vec![];
+    /// Returns the names of all exported [memories](crate::Memory) in the anonymous module.
+    pub fn mem_names_iter(&self) -> Option<Vec<String>> {
```

Comment:
`_iter` should be removed as above.

Reply:
Accepted. The issues has been fixed, please review the latest changes. Thanks a lot!